OpenRA.Mods.Common/Warheads/FireClusterWarhead.cs

```diff
@@ -0,0 +1,92 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2016 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.GameRules;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Warheads
+{
+	public class FireClusterWarhead : Warhead, IRulesetLoaded<WeaponInfo>
+	{
+		[WeaponReference, FieldLoader.Require]
+		[Desc("Has to be defined in weapons.yaml as well.")]
+		public readonly string Weapon = null;
+
+		[Desc("The range of the cells where the weapon should be fired.",
+			"A single value means radius, two values mean rectangle.")]
+		public readonly int[] Range = new int[] { 1 };
+
+		WeaponInfo weapon;
+
+		public void RulesetLoaded(Ruleset rules, WeaponInfo info)
+		{
+			weapon = rules.Weapons[Weapon.ToLowerInvariant()];
```

Comment:
Please use `TryGetValue` and throw a friendlier exception if the weapon doesn't exist (cf. https://github.com/OpenRA/OpenRA/commit/4f798ede3c34cc249e6ab26aafa452aebfdec6f1).